File: server/src/main/java/de/ingrid/igeserver/api/LoginApiController.java
package de.ingrid.igeserver.api;


import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(path="/api")
public class LoginApiController implements LoginApi {

    public ResponseEntity<Void> login(String username, String password) {
        // do some magic!
        return new ResponseEntity<Void>(HttpStatus.OK);
    }
}

File: server/src/main/java/de/ingrid/igeserver/api/LoginApiController.kt
package de.ingrid.igeserver.api

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = ["/api"])
class LoginApiController : LoginApi {
    override fun login(username: String?, password: String?): ResponseEntity<Void> {
        // do some magic!
        return ResponseEntity(HttpStatus.OK)
    }
}